src/app/core/permissions/user-role.guard.ts

```diff
@@ -0,0 +1,35 @@
+import { Injectable } from "@angular/core";
+import { ActivatedRouteSnapshot, CanActivate } from "@angular/router";
+import { SessionService } from "../session/session-service/session.service";
+import { ConfigService } from "../config/config.service";
+import {
+  PREFIX_VIEW_CONFIG,
+  ViewConfig,
+} from "../view/dynamic-routing/view-config.interface";
+
+@Injectable()
+/**
+ * A guard that checks the roles of the current user against the permissions of the route config.
+ */
+export class UserRoleGuard implements CanActivate {
+  constructor(
+    private sessionService: SessionService,
+    private configService: ConfigService
+  ) {}
+
+  canActivate(route: ActivatedRouteSnapshot): boolean {
+    const config = this.configService.getConfig<ViewConfig>(
```

Comment:
rather than loading the config here we should simply be able to use `route.data` as the RouterService assigns the config object to each route during setup.

Reply:
Currently the `permittedUserRoles` object is not inside the `config` which is stored in `route.data` therefore it's not possible to get the permission information from `route.data`. Of course the `permittedUserRoles` could be moved into the `config` object but I find it a quite good separation to have the `ViewConfig` interface which handles all the organizational stuff (component, lazyLoaded, permissions,...) while the `config` object only contains the information that are needed by the component that is displayed.